front/src/common/Map/Settings/MapSettingsSpeedLimits.tsx

```diff
@@ -51,18 +52,20 @@ const FormatSwitch = ({ name, icon: IconComponent, color = '', disabled }: Forma
     );
   };
 
-  const dispatchSetSpeedLimitsTags = (item?: string) => {
-    if (item)
+  const dispatchSetSpeedLimitsTags = (item?: string | null) => {
+    if (item) {
+      const newTag = item !== DEFAULT_SPEED_LIMIT_TAG ? item : null;
       dispatch(
         updateLayersSettings({
           ...layersSettings,
-          speedlimittag: item,
+          speedlimittag: newTag,
         })
       );
+    }
   };
 
   useEffect(() => {
-    if (tagsList) setSpeedLimitsTags([t('noSpeedLimitByTag').toString(), ...tagsList]);
+    if (tagsList) setSpeedLimitsTags([DEFAULT_SPEED_LIMIT_TAG, ...tagsList]);
     else setSpeedLimitsTags(undefined);
   }, [tagsList]);
```

Comment:
You can replace this useEffect by a useMemo const
```suggestion
  const speedLimitsTags = useMemo(
    () => (tagsList ? [DEFAULT_SPEED_LIMIT_TAG, ...tagsList] : undefined),
    [tagsList]
  );
```